diff --git a/dist/bspatch.mjs b/dist/bspatch.mjs
index a8a953becd4228c250aab15b4b9d281598f3464e..d3a9e650d05ee03d27a656fc08fb34b890435ded 100644
--- a/dist/bspatch.mjs
+++ b/dist/bspatch.mjs
@@ -5,14 +5,14 @@ var loadBspatch = (() => {
   return (
 async function(moduleArg = {}) {
 
-var d=moduleArg,aa,ba;d.ready=new Promise((a,b)=>{aa=a;ba=b});var ca=Object.assign({},d),da=[],ea="./this.program",fa=(a,b)=>{throw b;},ia="object"==typeof window,k="function"==typeof importScripts,p="object"==typeof process&&"object"==typeof process.versions&&"string"==typeof process.versions.node,t="",ja,ka,la;
+var d=moduleArg,aa,ba;d.ready=new Promise((a,b)=>{aa=a;ba=b});var ca=Object.assign({},d),da=[],ea="./this.program",fa=(a,b)=>{throw b;},ia="object"==typeof window,k="function"==typeof importScripts,p="object"==typeof process&&"object"==typeof process.versions&&"string"==typeof process.versions.node&&!("object"==typeof navigator&&"Cloudflare-Workers"===navigator.userAgent),t="",ja,ka,la;
 if(p){const {createRequire:a}=await import("module");var require=a(import.meta.url),fs=require("fs"),ma=require("path");k?t=ma.dirname(t)+"/":t=require("url").fileURLToPath(new URL("./",import.meta.url));ja=(b,c)=>{b=na(b)?new URL(b):ma.normalize(b);return fs.readFileSync(b,c?void 0:"utf8")};la=b=>{b=ja(b,!0);b.buffer||(b=new Uint8Array(b));return b};ka=(b,c,e,f=!0)=>{b=na(b)?new URL(b):ma.normalize(b);fs.readFile(b,f?void 0:"utf8",(g,h)=>{g?e(g):c(f?h.buffer:h)})};
 !d.thisProgram&&1<process.argv.length&&(ea=process.argv[1].replace(/\\/g,"/"));da=process.argv.slice(2);fa=(b,c)=>{process.exitCode=b;throw c;}}else if(ia||k)k?t=self.location.href:"undefined"!=typeof document&&document.currentScript&&(t=document.currentScript.src),_scriptDir&&(t=_scriptDir),t.startsWith("blob:")?t="":t=t.substr(0,t.replace(/[?#].*/,"").lastIndexOf("/")+1),ja=a=>{var b=new XMLHttpRequest;b.open("GET",a,!1);b.send(null);return b.responseText},k&&(la=a=>{var b=new XMLHttpRequest;b.open("GET",
 a,!1);b.responseType="arraybuffer";b.send(null);return new Uint8Array(b.response)}),ka=(a,b,c)=>{var e=new XMLHttpRequest;e.open("GET",a,!0);e.responseType="arraybuffer";e.onload=()=>{200==e.status||0==e.status&&e.response?b(e.response):c()};e.onerror=c;e.send(null)};var oa=d.print||console.log.bind(console),u=d.printErr||console.error.bind(console);Object.assign(d,ca);ca=null;"object"!=typeof WebAssembly&&v("no native wasm support detected");var pa,qa=!1,ra,w,sa,ta,y,A,ua;
 function va(){var a=pa.buffer;d.HEAP8=w=new Int8Array(a);d.HEAP16=ta=new Int16Array(a);d.HEAPU8=sa=new Uint8Array(a);d.HEAPU16=new Uint16Array(a);d.HEAP32=y=new Int32Array(a);d.HEAPU32=A=new Uint32Array(a);d.HEAPF32=new Float32Array(a);d.HEAPF64=new Float64Array(a);d.HEAP64=ua=new BigInt64Array(a);d.HEAPU64=new BigUint64Array(a)}var wa=[],xa=[],ya=[],za=[],D=0,Aa=null,Ba=null;function Ca(){D--;if(0==D&&(null!==Aa&&(clearInterval(Aa),Aa=null),Ba)){var a=Ba;Ba=null;a()}}
 function v(a){a="Aborted("+a+")";u(a);qa=!0;ra=1;a=new WebAssembly.RuntimeError(a+". Build with -sASSERTIONS for more info.");ba(a);throw a;}var Da=a=>a.startsWith("data:application/octet-stream;base64,"),na=a=>a.startsWith("file://"),E;if(d.locateFile){if(E="bspatch.wasm",!Da(E)){var Ea=E;E=d.locateFile?d.locateFile(Ea,t):t+Ea}}else E=(new URL("bspatch.wasm",import.meta.url)).href;function Fa(a){if(la)return la(a);throw"both async and sync fetching of the wasm failed";}
 function Ga(a){if(ia||k){if("function"==typeof fetch&&!na(a))return fetch(a,{credentials:"same-origin"}).then(b=>{if(!b.ok)throw`failed to load wasm binary file at '${a}'`;return b.arrayBuffer()}).catch(()=>Fa(a));if(ka)return new Promise((b,c)=>{ka(a,e=>b(new Uint8Array(e)),c)})}return Promise.resolve().then(()=>Fa(a))}function Ha(a,b,c){return Ga(a).then(e=>WebAssembly.instantiate(e,b)).then(c,e=>{u(`failed to asynchronously prepare wasm: ${e}`);v(e)})}
-function Ia(a,b){var c=E;return"function"!=typeof WebAssembly.instantiateStreaming||Da(c)||na(c)||p||"function"!=typeof fetch?Ha(c,a,b):fetch(c,{credentials:"same-origin"}).then(e=>WebAssembly.instantiateStreaming(e,a).then(b,function(f){u(`wasm streaming compile failed: ${f}`);u("falling back to ArrayBuffer instantiation");return Ha(c,a,b)}))}function Ja(a){this.name="ExitStatus";this.message=`Program terminated with exit(${a})`;this.status=a}
+function Ia(a,b){if(d.instantiateWasm)try{return Promise.resolve(d.instantiateWasm(a,b))}catch(c){return Promise.reject(c)}var c=E;return"function"!=typeof WebAssembly.instantiateStreaming||Da(c)||na(c)||p||"function"!=typeof fetch?Ha(c,a,b):fetch(c,{credentials:"same-origin"}).then(e=>WebAssembly.instantiateStreaming(e,a).then(b,function(f){u(`wasm streaming compile failed: ${f}`);u("falling back to ArrayBuffer instantiation");return Ha(c,a,b)}))}function Ja(a){this.name="ExitStatus";this.message=`Program terminated with exit(${a})`;this.status=a}
 var Ka=a=>{for(;0<a.length;)a.shift()(d)},La=(a,b)=>{for(var c=0,e=a.length-1;0<=e;e--){var f=a[e];"."===f?a.splice(e,1):".."===f?(a.splice(e,1),c++):c&&(a.splice(e,1),c--)}if(b)for(;c;c--)a.unshift("..");return a},G=a=>{var b="/"===a.charAt(0),c="/"===a.substr(-1);(a=La(a.split("/").filter(e=>!!e),!b).join("/"))||b||(a=".");a&&c&&(a+="/");return(b?"/":"")+a},Ma=a=>{var b=/^(\/?|)([\s\S]*?)((?:\.{1,2}|[^\/]+?|)(\.[^.\/]*|))(?:[\/]*)$/.exec(a).slice(1);a=b[0];b=b[1];if(!a&&!b)return".";b&&=b.substr(0,
 b.length-1);return a+b},H=a=>{if("/"===a)return"/";a=G(a);a=a.replace(/\/$/,"");var b=a.lastIndexOf("/");return-1===b?a:a.substr(b+1)},Na=(...a)=>G(a.join("/")),I=(a,b)=>G(a+"/"+b),Oa=()=>{if("object"==typeof crypto&&"function"==typeof crypto.getRandomValues)return c=>crypto.getRandomValues(c);if(p)try{var a=require("crypto");if(a.randomFillSync)return c=>a.randomFillSync(c);var b=a.randomBytes;return c=>(c.set(b(c.byteLength)),c)}catch(c){}v("initRandomDevice")},Pa=a=>(Pa=Oa())(a),K=(...a)=>{for(var b=
 "",c=!1,e=a.length-1;-1<=e&&!c;e--){c=0<=e?a[e]:J.cwd();if("string"!=typeof c)throw new TypeError("Arguments to path.resolve must be strings");if(!c)return"";b=c+"/"+b;c="/"===c.charAt(0)}b=La(b.split("/").filter(f=>!!f),!c).join("/");return(c?"/":"")+b||"."},Qa=(a,b)=>{function c(h){for(var m=0;m<h.length&&""===h[m];m++);for(var n=h.length-1;0<=n&&""===h[n];n--);return m>n?[]:h.slice(m,n-m+1)}a=K(a).substr(1);b=K(b).substr(1);a=c(a.split("/"));b=c(b.split("/"));for(var e=Math.min(a.length,b.length),
diff --git a/package.json b/package.json
index bfda854c49c7eac628fc8c82680d111950225d25..f9d9b826e0bc31cc81319bc6739e7fae11d1639a 100644
--- a/package.json
+++ b/package.json
@@ -7,7 +7,8 @@
   "exports": {
     ".": "./dist/main.mjs",
     "./bsdiff": "./dist/bsdiff.mjs",
-    "./bspatch": "./dist/bspatch.mjs"
+    "./bspatch": "./dist/bspatch.mjs",
+    "./bspatch.wasm": "./dist/bspatch.wasm"
   },
   "files": [
     "dist",
